src/pip/_internal/commands/list.py

```diff
@@ -145,7 +146,8 @@ def run(self, options, args):
         # from current directory
         paths = options.path
         if paths is None:
-            paths = [item for item in sys.path if item and item != os.getcwd()]
+            paths = [item for item in sys.path if item and
+                     normalize_path(item) != normalize_path(os.getcwd())]
```

Comment:
Do we actually want to test for `os.getcwd()`? Or is skipping the empty `sys.path` item added by python sufficient? Are there circumstances where cwd is added to sys.path (other than the user adding it, in which case it is desirable to not skip it).

Reply:
Hi @sbidoul , 

According to the documentation on [python -m](https://docs.python.org/3/using/cmdline.html#cmdoption-m), `As with the -c option, the current directory will be added to the start of sys.path.` .  

Hence I am checking for `os.cwd()` explicitly

Reply:
I'm not sure if you've read this:  https://github.com/pypa/pip/pull/7810#pullrequestreview-383321074

Reply:
No worries @McSinyx , so is my reasoning correct and in line with the question asked?

Reply:
@McSinyx thanks for the heads up. I now see that sometimes we get `''` in sys.path and sometimes the absolute cwd. That is somewhat surprising and I'd be interested to understand why it is the case.

Nevertheless I'm asking myself if there are legitimate use cases where people manually add cwd to PYTHONPATH that this PR would break.

Reply:
Hi @sbidoul , 

So I this PR I am checking if the paths present in sys.path equals cwd for every path. I am not differentiating if the path was added by the command `python -m`, or a user added it themselves (I am not aware of any way to differentiate such a thing).

Also what would be the expected behaviour if the user themselves add path to cwd? Should they expect to see the package in cwd in `python -m pip list`, but they won't since we are removing it?

One way I could think of is we count how many occurrences of cwd are in `sys.path` and only remove the first occurrence which is added by python, and leave the others which might be user added. What do you think?

Reply:
Though, if user would want to include the CWD, which should not be the default behavior, then s/he would use `--path` option to list the packages of CWD, which we are already using, i.e. `if paths is None`, then cwd is excluded from `sys.path` else `paths` would contain the path provided by the user.

So, I think this should work. But this is still weird behavior, that `sys.path` occasionally contains the absolute path of CWD without explicitly mentioning. I was only aware of it adding an empty string `''` at the start of it(which corresponds to the cwd already).

Reply:
Though, if a user would want to check for packages present in CWD, then s/he can use the `--path` option, which is what exactly is being performed here i.e. `if paths is None`, then only we are removing the CWD from `sys.path`(which should be the default behavior). Otherwise, `paths` will be populated with the user provided paths and that'd be used in that case.

So, it does seem to work. However, it is actually weird that `sys.path` occasionally contains the absolute path apart from `''`(which actually refers to CWD). :|